include/hueplusplus/Condition.h

```diff
@@ -0,0 +1,168 @@
+/**
+    \file Condition.h
+    Copyright Notice\n
+    Copyright (C) 2020  Jan Rogall          	- developer\n
+
+    This file is part of hueplusplus.
+
+    hueplusplus is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    hueplusplus is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with hueplusplus.  If not, see <http://www.gnu.org/licenses/>.
+**/
+
+#ifndef INCLUDE_HUEPLUSPLUS_CONDITION_H
+#define INCLUDE_HUEPLUSPLUS_CONDITION_H
+
+#include "TimePattern.h"
+
+#include "json/json.hpp"
+
+namespace hueplusplus
+{
+//! \brief Condition for a Rule
+//!
+//! The condition checks whether a resource attribute (usually a Sensor value) matches the
+//! specified Operator.
+//!
+//! Conditions from sensors can be created more easily using the makeCondition() helper functions.
+class Condition
+{
+public:
+    //! \brief Specifies which operation is used to check the condition
+    enum class Operator
+    {
+        eq, //!< Attribute is equal to specified value (for bool and int)
+        gt, //!< Attribute is greater than specified value (for int)
+        lt, //!< Attribute is less than specified value (for int)
+        dx, //!< Attribute has changed (no value given)
+        ddx, //!< Delayed attribute has changed (no value given)
+        stable, //!< Stable for a given time. Does not trigger a rule change
+        notStable, //!< Not stable for a given time. Does not trigger a rule change
+        in, //!< Time is in the given interval (triggered on start time, local time)
+        notIn //!< Time is not in the interval (triggered on end time, local time)
```

Comment:
```suggestion
        NOT_IN //!< Time is not in the interval (triggered on end time, local time)
```
Please also change eq to EQ, gt to GT, lt to LT, dx to DX, ddx to DDX, stable to STABLE, notStable to NOT_STABLE and in to IN